kmk/kmk_keyboard.py

```diff
@@ -67,32 +68,32 @@ class KMKKeyboard:
     # real known fix yet other than turning off debug, but M4s have always been
     # tight on RAM so....
     def __repr__(self):
-        return (
-            'KMKKeyboard('
-            'debug_enabled={} '
-            'diode_orientation={} '
-            'matrix={} '
-            'unicode_mode={} '
-            '_hid_helper={} '
-            'keys_pressed={} '
-            'coordkeys_pressed={} '
-            'hid_pending={} '
-            'active_layers={} '
-            'timeouts={} '
-            ')'
-        ).format(
-            self.debug_enabled,
-            self.diode_orientation,
-            self.matrix,
-            self.unicode_mode,
-            self._hid_helper,
-            # internal state
-            self.keys_pressed,
-            self._coordkeys_pressed,
-            self.hid_pending,
-            self.active_layers,
-            self._timeouts,
-        )
+        txt = 'KMKKeyboard(\n'
+        line = '  '
+        for attr_name in [
+            'debug_enabled',
+            'diode_orientation',
+            'matrix',
+            'unicode_mode',
+            '_hid_helper',
+            # internal stat
+            'keys_pressed',
+            '_coordkeys_pressed',
+            'hid_pending',
+            'active_layers',
+            '_timeouts',
+        ]:
+            attr = getattr(self, attr_name)
+            part = f'{attr_name}={attr}, '
```

Comment:
I'm unsure how I feel about `getattr` in this context since this becomes less statically analyzable (similar to my vote in the other PR #461) - I love the premise of what this PR is getting at (especially as an intermediate step until a more robust debug framework is in place); would it be feasible to loop over the `self.*` attrs as references rather than strings to look up with `getattr`? perhaps

```python
for ref in [ self.debug_enabled, self.diode_orientation, ... ]:
```

Reply:
TL;DR: agree with @klardotsh, but won't block anything

`getattr` and `self.__getattr__` are exactly the same thing. The problem comes from the `attr_name`. You can put whatever in there, and no _static_ analysis will ever find out if it resolves to a valid attribute, because it's by design a _dynamic_ attribute reference. The only way to check for validity is to run the code, which in turn defeats the purpose of a static analysis.

More importantly: verbosity isn't necessarily a bad thing, it's about finding the sweet spot for human parsability. Writing code with the fewest possible symbols is very exciting and deserves it's own space and recognition. However: it's hardly conducive in project that relies on a community of contributers with a wide range of coding background and familiarity with the code base.
In this day and age (apologies for the platitude), programming is 20% writing your own and 80% reading other peoples code. It's important (and will silently be appreciated by others) to write code that can first and foremost be understood by people with less experience. Otherwise we'd be putting inline assembly everywhere.
To be clear: I agree that code should be as concise as possible, but not conciser (I think I'm misquoting Einstein).

To give that context to this PR:
I'd go even further than @klardotsh and suggest: make it an f-string with static newlines embedded. Then the output will look exactly like the string is formated in the code, which is nice. Put the potentially long attributes in seperate lines. Once those wrap you've got so much text on screen, that dynamic line wrapping isn't going to meaningfully improve readability anyway.
I won't be blocking this PR, it's an improvement to the status quo, even with the `getattr`. It's at a place where it's not going to do much harm, but be aware that it may get refactored at some point.

Reply:
Ok, I have updated to the fixed multiline way then.